core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -217,7 +226,13 @@ public PartitionSpec apply() {
         if (newName != null) {
           builder.add(field.sourceId(), field.fieldId(), newName, field.transform());
         } else {
-          builder.add(field.sourceId(), field.fieldId(), field.name(), field.transform());
+          // field name of void transform will be replaced by field name append with UUID if conflict is detected.
+          String partitionName = field.name();
+          if (isVoidTransform(field) && (renames.containsValue(partitionName) ||
+                  adds.stream().anyMatch(newField -> field.name().equals(newField.second())))) {
```

Comment:
Nit: indentation is off. Should be 4 spaces / 2 indents from the start of `if`.

Reply:
fix it in next commit